pkg/detectors/loggly/loggly.go

```diff
@@ -0,0 +1,95 @@
+package loggly
+
+import (
+        "context"
+        "fmt"
+        "net/http"
+        "regexp"
+        "strings"
+
+        "github.com/trufflesecurity/trufflehog/v3/pkg/common"
+        "github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+        "github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+        client = common.SaneHttpClient()
+        // Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+        domainPat = regexp.MustCompile(`\b([a-zA-Z0-9-]+\.loggly\.com)\b`)
+        keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"loggly"}) + `\b([a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+        return []string{"loggly"}
+}
+
+// FromData will find and optionally verify Loggly secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+        dataStr := string(data)
+
+        keyMatches := keyPat.FindAllStringSubmatch(dataStr, -1)
+        domainMatches := domainPat.FindAllStringSubmatch(dataStr, -1)
+
+        for _, match := range keyMatches {
+                if len(match) != 2 {
+                        continue
+                }
+                key := strings.TrimSpace(match[1])
+
+                for _, domainMatch := range domainMatches {
+                        if len(domainMatch) != 2 {
```

Comment:
@ankushgoel27 tests need to be updated to reflect the capture of 2 matches. Right now the tests are only supplying a sample key but leave out the domain.

Reply:
Sorry, Dont understand what you mean.

Reply:
fixed the loggly_test.go